technology/core-skills/tdd/inappropriate.md

```diff
@@ -1,11 +1,24 @@
 # Inappropriate TDD
 
-Graphical UIs are not suitable for TDD, they are usually defined in a markup language which can only be asserted for correctness by visually inspecting the rendered output.
+## Validation of correctness cannot be obtained via an automated check.
 
-Configurations are not suitable for TDD, their correctness can only be ascertained by testing the configuration in conjunction with whatever they are configuring.
+Sometimes you can find a proxy for "correctness" which will make you feel like TDD is possible.
+If you aren't feeling confident that your tests are guarding against regression, this is a signal that you shouldn't be using TDD.
 
-State machines are configuration? Unit testing is still important, but the discipline of TDD is not.
+Graphical UIs are an example of software not suitable for TDD, they are usually defined in a markup language which can only be asserted for correctness by visually inspecting the rendered output.
```

Comment:
Might not be wholly relevant to mention this extra context so feel free to disregard, but you could argue that while the user interactions and visual correctness can't be test-driven, you can test-drive, for instance, a HTML document that's programmatically built (e.g., asserting an element contains the correct `aria` tags given certain conditions); related to subcutaneous testing, I guess